services/frontend-service/src/ui/TopAppBar/TopAppBar.tsx

```diff
@@ -17,21 +17,16 @@ Copyright 2021 freiheit.com*/
 import { MDCTopAppBar } from '@material/top-app-bar';
 
 import { Textfield } from '../components/textfield';
-import { useCallback, useEffect, useRef } from 'react';
+import React, { useCallback, useEffect, useRef, useState } from 'react';
 import { SideBar } from '../SideBar/SideBar';
 import { Button } from '../components/button';
 
 export const TopAppBar: React.FC = () => {
     const control = useRef<HTMLDivElement>(null);
     const MDComponent = useRef<MDCTopAppBar>();
+    const [sideBar, showSideBar] = useState(false);
 
-    const displaySideBar = useCallback(() => {
-        const sideBarRef = document.getElementsByClassName('mdc-drawer--hidden')[0];
-        if (sideBarRef?.classList.contains('mdc-drawer--hidden')) {
-            sideBarRef?.classList.remove('mdc-drawer--hidden');
-            sideBarRef?.classList.add('mdc-drawer--displayed');
-        }
-    }, []);
+    const displaySideBar = useCallback(() => showSideBar(!sideBar), [sideBar]);
```

Comment:
maybe call this `toggleSidebar` as it toggles the state of the sidebar

Reply:
Please use this:

```
useCallback(() => showSideBar(old => !old), [showSideBar]);
```

otherwise you will get race-conditions.

Reply:
Alright, fixed!